filebeat/input/filestream/prospector.go

```diff
@@ -129,46 +129,46 @@ func (p *fileProspector) Run(ctx input.Context, s loginp.StateMetadataUpdater, h
 			}
 
 			src := p.identifier.GetSource(fe)
-			log = loggerWithEvent(log, fe, src)
+			evtLog := loggerWithEvent(log, fe, src)
```

Comment:
My recommendation would be to have a separate function, so we don't have too many loggers within the same context.